moped-editor/src/views/projects/projectView/ProjectStatusBadge.js

```diff
@@ -0,0 +1,243 @@
+      color: white,
+      background: backgroundColors.success,
+      icon: PlayCircleOutlineOutlinedIcon,
+    },
+    design: {
+      color: white,
+      background: backgroundColors.success,
+      icon: PlayCircleOutlineOutlinedIcon,
+    },
+    "pre-construction": {
+      color: white,
+      background: backgroundColors.success,
+      icon: PlayCircleOutlineOutlinedIcon,
+    },
+    "bid/award/execution": {
+      color: white,
+      background: backgroundColors.success,
+      icon: PlayCircleOutlineOutlinedIcon,
+    },
+    "construction-ready": {
+      color: white,
+      background: backgroundColors.success,
+      icon: PlayCircleOutlineOutlinedIcon,
+    },
+    construction: {
+      color: white,
+      background: backgroundColors.success,
+      icon: PlayCircleOutlineOutlinedIcon,
+    },
+    "post-construction": {
+      color: white,
+      background: backgroundColors.success,
+      icon: PlayCircleOutlineOutlinedIcon,
+    },
+    potential: {
+      color: primary,
+      background: backgroundColors.warning,
+      icon: RemoveCircleOutlineOutlinedIcon,
+    },
+    canceled: {
+      color: white,
+      background: backgroundColors.error,
+      icon: CancelOutlinedIcon,
+    },
+    "on hold": {
+      color: primary,
+      background: backgroundColors.default,
+      icon: PauseCircleOutlineOutlinedIcon,
+    },
+    "on-hold": {
+      color: primary,
+      background: backgroundColors.default,
+      icon: PauseCircleOutlineOutlinedIcon,
+    },
+    complete: {
+      color: white,
+      background: backgroundColors.info,
+      icon: CheckCircleOutlineOutlinedIcon,
+    },
+    default: {
+      color: primary,
+      background: defaultBackgroundColor,
+      icon: defaultIcon,
+    },
+  };
+
+  return styleMapping?.[phase] ?? styleMapping.default;
+};
+
+/**
+ * Picks the right color based on the status-phase combination
+ */
+const useFontColorStyles = makeStyles(theme => ({
+  root: {
+    // Find text color
+    color: props => getStyle(theme, (props?.phase ?? "").toLowerCase()).color,
+  },
+}));
+
+/**
+ * Picks the right chip color based on status-phase combination
+ */
+const useChipStyles = makeStyles(theme => ({
+  root: {
+    marginLeft: "1.5rem",
+    fontWeight: "500",
+    fontSize: "16px",
+    borderRadius: "2rem",
+    height: "2.5rem",
+    padding: ".5rem",
+
+    // Find background color
+    backgroundColor: props => getStyle(theme, props?.phase ?? "").background,
+  },
+}));
+
+/**
+ * Renders a chip
+ * @param {number} status - The status id of the current project
+ * @param {string} phase - The current phase name of the project
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const ProjectStatusBadge = ({ status, phase }) => {
+  /**
+   * Returns the label given a status-phase combination
+   * @param status
+   * @param phase
+   * @returns {string}
+   */
+  const getComponentMapName = (status, phase) => {
+    switch (status) {
+      case 3:
```

Comment:
perhaps we should extract these magic numbers to a settings file so they're easier to track down if we make a breaking change down the road?

Reply:
Ditto. Didn't want to optimize too soon, just wanted to make sure I delivered the feature correctly first.